js/ui/appDisplay.js

```diff
@@ -667,11 +676,25 @@ var AllView = class AllView extends BaseAppView {
         };
     }
 
+    _ensureHackAppIcon() {
+        if (this._hackAppIcon)
+            return;
+
+        this._hackAppIcon = new HackAppIcon(this);
+        this._hackAppItem = {
+            get_name: () => { return this._hackAppIcon.getName(); }
```

Comment:
I assume the arrow function is for intentionally capturing `this`? I would advise against doing that, it's confusing and easy to misunderstand. If you have to do it, clarify it with a comment.

Reply:
This is a copy of the same code for the `AppCenterIcon` that's just above.

What do you suggest as a replacement of this? The first thing that comes to my mind is to assign directly to the `getName` method, something like this:

```
this._hackAppItem = { get_name: this._hackAppIcon.getName };
```

Reply:
I would use `bind` which makes it more obvious that the `this` is not what you think when you are calling `get_name`:

```js
get_name: this._hackAppIcon.getName.bind(this),
```

Reply:
done